keep DisableContentSha256 as defaut value during mc perf test

moting9 opened the pull request:
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
when I use 'mc perf' to test MinIO cluster performance, I found GET result by mc is similar as that by warp, but PUT result by mc is better than that by warp.  I noticed mc put has disabled sha256 by default.

## Motivation and Context
The change is to keep DisableContentSha256 in PutObjectOptions as default value in mc perf test  to align with warp

## How to test this PR?
 ./mc support perf object  myminio/ 

 ./mc admin trace -v myminio     
original PUT request header is:
 X-Amz-Content-Sha256: UNSIGNED-PAYLOAD
new PUT request header is 
 X-Amz-Content-Sha256: STREAMING-AWS4-HMAC-SHA256-PAYLOAD

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

moting9 committed:
Make DisableContentSha256 configurable in mc perf test

harshavardhana commented on cmd/perf-tests.go:
```diff
@@ -95,7 +95,6 @@ func selfSpeedTest(ctx context.Context, opts speedTestOpts) (SpeedTestResult, er
 	userMetadata[globalObjectPerfUserMetadata] = "true" // Bypass S3 API freeze
 	popts := minio.PutObjectOptions{
 		UserMetadata:         userMetadata,
-		DisableContentSha256: true,
```
This needs to come as a configurable parameter @moting9 - I ended up changing warp https://github.com/minio/warp/pull/292 to have `--disable-sha256-payload`

harshavardhana committed:
allow pre-allocating buffers to reduce frequent GCs during growth (#18686)

This PR also increases per node bpool memory from 1024 entries
to 2048 entries; along with that, it also moves the byte pool
centrally instead of being per pool.

vadmeste committed:
prom: Add online and healing drives metrics per erasure set (#18700)

harshavardhana committed:
send proper IPv6 names avoid bracketing notation (#18699)

Following policies if present

```
       "Condition": {
         "IpAddress": {
            "aws:SourceIp": [
              "54.240.143.0/24",
               "2001:DB8:1234:5678::/64"
             ]
          }
        }
```

And client is making a request to MinIO via IPv6 can
potentially crash the server.

Workarounds are turn-off IPv6 and use only IPv4

harshavardhana committed:
rename object_size -> block_size for cache subsystem (#18694)

vadmeste committed:
tier: Allow edit of the new Azure and AWS auth params (#18690)

Allow editing for the service principal credentials from Azure
and the web identity token for AWS;

Also, more validation of input parameters.

harshavardhana committed:
fix: reject clients that do not send proper payload (#18701)

shtripat committed:
Graph cluster wide where applicable (#18705)

Graph the maximum value reported across nodes at cluster
level for applicable scenarios.

Signed-off-by: Shubhendu Ram Tripathi <shubhendu@minio.io>

vadmeste committed:
server-info: Avoid initializing audit/log http/kafka targets (#18703)

This can cause unnecessary ServerInfo() call delay.

donatello committed:
Fix precendence bug in S3Select SQL IN clauses (#18708)

Fixes a precendence issue in SQL Select where `a in b and c = 3` was parsed as `a
in (b and c = 3)`.

Fixes #18682

minio-trusted committed:
Update yaml files to latest version RELEASE.2023-12-23T07-19-11Z

harshavardhana committed:
upgrade Go mod dependencies (#18710)
